administrator/components/com_media/resources/scripts/components/browser/browser.vue

```diff
@@ -1,5 +1,6 @@
 <template>
     <div>
+        <media-toolbar :search="query" @newQuery="query = $event"></media-toolbar>
```

Comment:
It is not necessary to make media-toolbar a child of media-browser which is technically wrong.
One possible solution here to keep a state variable called search and when you are typing just mutate the value of it. Then you can get it from `browser` as a computed property. When you are committing mutations it will be propagate to the browser. So you can remove the extra events fired here and decouple things better.